### ISSUE TITLE
json to json schema -- Go to declaration/type declaration

### ISSUE THREAD

**Issue by neopaf** [microsoft/vscode#138546]
- VS Code Version: Version: 1.62.3 (Universal)
- OS: Darwin x64 19.6.0

Steps to Reproduce:

[VSC bug 1.zip](https://github.com/microsoft/vscode/files/7663115/VSC.bug.1.zip)
1. open bug1.json from attached minimal project
2. position cursor over field1 (line 2, column 7)
3. Go to Definition

Expected: jump to bug1.schema.json, line 5, column 5.
Actual: nothing happens.

Maybe I expect too much?
I'd like to edit both example and its specification file in parallel and am very interested in cross-navigation

**Comment by aeschli** [comment]
We don't have that JSON document to schema functionality.
It would be nice to have, if you are interested to work on it... :-)

Since a while the JSON language extension does not have a Definition provider anymore, so you should not see the `Go to Definition` command in the context menu. Maybe another extension provides it?

![image](https://user-images.githubusercontent.com/6461412/145019529-bf2b977f-9d33-4efa-a608-477211ce7a76.png)

**Comment by neopaf** [comment]
I guess it was more of a wishful thinking on my side than actual experiment.
I was command+clicking the field, some reason assuming I should get to declaration.

when I'm developing a new JSON schema, I like to do that hand-in-hand with real-world example.
And design goes from Example: adding code there, and would expect some validation (ideally: all invalid blocks red), AND cross-navigation (from usage to definition and back).

**Comment by aeschli** [comment]
I fully agree that this would be a great experience.

**Comment by thernstig** [comment]
It would be a great experience! Upvoted. Upvote it you as well @neopaf as otherwise it will be closed. And ask your colleagues to upvote.

@aeschli I don't have the `Go to Definition`, but I do see `Peek -> Peek Definition`. Is that a miss in the JSON language extension? Separate bug?

**Comment by neopaf** [comment]
not sure how to do that, @thernstig, I can't seem to have relevant UI...
